### PR TITLE
REST API: Add algod block hash endpoint, add indexer block header-only param.

### PR DESCRIPTION
**Pull request by winder**
REST API changes and test implementations.

Tests are passing locally, CI is pending https://github.com/algorand/algorand-sdk-testing/pull/245

### CODE REVIEW

**Review by tzaffi — COMMENTED** [review]
seems good to me. We should base the changes in the harness script off the definitive version of it in https://github.com/algorand/go-algorand-sdk/pull/421

**Review by tzaffi — APPROVED** [review]
Approving with the understanding that tests should pass before merging. I also recommend someone more experienced in Javascript have a look.

**Comment by barnjamin** [comment]
@winder i pr'd a fmt tweak that made the linter happy in https://github.com/algorand/js-algorand-sdk/pull/667

**Review by algochoi — APPROVED** [review]

**Pull request merged**
